### PR TITLE
LIVE-90-89: Add WebRTC grid component, HLS component, refactor frontend

### PR DESCRIPTION
**Pull request by LVala**
Add WebRTC component, add HLS stream component, refactor css files.

### CODE REVIEW

File: assets/js/utils/modePanelUtils.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    const updateParticipantsNumber = () => {
+      setPresentersNumber(presence.list().filter((elem) => elem.metas[0].is_presenter).length);
+    };
+
+    const upadteAmIPresenter = () => {
+      let newAmIPresenter = false;
+      presence.list((name: string, metas: any) => {
+        if (metas.metas[0].is_presenter && name == clientName) newAmIPresenter = true;
+      });
```
Comment (Rados13) [inline assets/js/utils/modePanelUtils.ts:20]: What do you think about t
```
presence.list().find((elem) => elem.metas[0].is_presenter && elem.name == clientName) !== undefined
```

File: assets/js/pages/Event.tsx [hunk 2]
```diff
@@ -57,21 +56,11 @@ const Event = () => {
...
-          <div className="Stream">
-            <PresenterStreamArea clientName={name} eventChannel={eventChannel} />
-            <HLSPlayer eventChannel={eventChannel} />
-          </div>
-        </div>
-        <div className="Participants">
-          <ParticipantsList username={name} isModerator={true} eventChannel={eventChannel} />
-        </div>
+        <StreamArea clientName={name} eventChannel={eventChannel} />
+        <ParticipantsList clientName={name} isModerator={true} eventChannel={eventChannel} />
```
Comment (Rados13) [inline assets/js/pages/Event.tsx:63]: Are you sure that `isModerator` should be always true?

File: assets/js/components/StreamArea.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+const StreamArea = ({ clientName, eventChannel }: StreamAreaProps) => {
+  const [mode, setMode] = useState<Mode>("hls");
+  const [hlsUrl, setHlsUrl] = useState<string>("");
+  const [presenterName, setPresenterName] = useState<string>("");
+
+  const addHlsUrl = (message: { name: string; playlist_idl: string }): void => {
+    const link = window.location.href.split("event")[0] + "video/";
+    if (message.playlist_idl) {
+      setHlsUrl(link + message.playlist_idl + "/video.m3u8");
```
Comment (Rados13) [inline assets/js/components/StreamArea.tsx:22]: Use template literals.

File: assets/js/components/StreamArea.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        setMode={setMode}
+        presenterName={presenterName}
+        eventChannel={eventChannel}
+        clientName={clientName}
+      />
+      <div className="Stream">
+        {mode == "hls" && <HlsPlayer hlsUrl={hlsUrl} presenterName={presenterName} />}
+        <div className={mode == "hls" ? "Hidden" : ""}>
+          <PresenterStreams clientName={clientName} eventChannel={eventChannel} setMode={setMode} />
+        </div>
```
Comment (Rados13) [inline assets/js/components/StreamArea.tsx]: Could we make for consistency it like this?
```js
{ mode != "hls" && <PresenterStreams clientName={clientName} eventChannel={eventChannel} setMode={setMode} />}
```

File: lib/membrane_live/event.ex [hunk 5]
```diff
@@ -169,7 +169,7 @@ defmodule MembraneLive.Event do
 
     state =
       put_in(state, [:playlist_idls, peer.id], %{
-        name: peer.metadata["displayName"],
+        name: peer.metadata["name"],
```
Comment (Rados13) [inline lib/membrane_live/event.ex:172]: Are you sure about this change?

File: assets/js/components/ParticipantsList.tsx [hunk 6]
```diff
@@ -30,16 +27,20 @@ type ParticipantProps = {
...
   eventChannel: any;
 };
 
-const ModeratorMenu = ({ username, name, isPresenter, eventChannel }: ModeratorMenuProps) => {
+const ModeratorMenu = ({ clientName, name, isPresenter, eventChannel }: ModeratorMenuProps) => {
+  const style = getComputedStyle(document.body);
+  const fontColor = style.getPropertyValue("--font-dark-color");
+
   const handleClick = (e: any) => {
+    console.log(e.target.value);
```
Comment (Rados13) [inline assets/js/components/ParticipantsList.tsx:40]: Do we need this?

File: assets/js/components/ParticipantsList.tsx [hunk 7]
```diff
@@ -30,16 +27,20 @@ type ParticipantProps = {
...
-  username: string;
+  clientName: string;
   isModerator: boolean;
   eventChannel: any;
 };
 
-const ModeratorMenu = ({ username, name, isPresenter, eventChannel }: ModeratorMenuProps) => {
+const ModeratorMenu = ({ clientName, name, isPresenter, eventChannel }: ModeratorMenuProps) => {
+  const style = getComputedStyle(document.body);
+  const fontColor = style.getPropertyValue("--font-dark-color");
```
Comment (Rados13) [inline assets/js/components/ParticipantsList.tsx:37]: I think as we are using in a few places we could move this to a function, which would return `fontColor`.

File: assets/js/components/ParticipantsList.tsx [hunk 8]
```diff
@@ -30,16 +27,20 @@ type ParticipantProps = {
...
+
   const handleClick = (e: any) => {
+    console.log(e.target.value);
     if (e.target.value === "Set as a presenter") {
       const link = "private:" + window.location.pathname.split("/")[2] + ":";
-      eventChannel.push("presenter_prop", { moderator: link + username, presenter: link + name });
+      eventChannel.push("presenter_prop", { moderator: link + clientName, presenter: link + name });
     } else {
       const link = "private:" + window.location.pathname.split("/")[2] + ":";
```
Comment (Rados13) [inline assets/js/components/ParticipantsList.tsx:45]: You can move link variable outside this `if ... else`.

File: assets/js/components/Header.tsx [hunk 9]
```diff
@@ -23,40 +36,50 @@ const Header = ({ name, eventChannel }: HeaderProps) => {
...
         <button className="CopyButton" onClick={handleCopyButton}>
-          <img src={copyIcon} />
+          <Copy />
         </button>
       </div>
       <div className="User">
-        <AiOutlineUser className="UserIcon"></AiOutlineUser>
+        <User1 className="UserIcon" />
         <div className="UserName"> {name}</div>
       </div>
```
Comment (Rados13) [inline assets/js/components/Header.tsx:81]: Shouldn't we handle here image from google account?

File: assets/js/components/ControlPanel.tsx [hunk 10]
```diff
@@ -147,13 +192,36 @@ const ControlPanel = ({ clientName, webrtc, playerCallback }: ControlPanelProps)
...
+      <div className="ControlPanel">
+        <GenericButton icon={<Settings className="PanelButton" />} onClick={onOpen} />
+        <div className="CenterIcons">
+          {getMuteButton("video", Cam, CamDisabled)}
+          {getMuteButton("audio", Microphone, MicrophoneDisabled)}
+          <GenericButton
+            icon={
+              <PhoneDown
+                className="DisconnectButton"
+                color={style.getPropertyValue("--bg-light-color-1")}
```
Comment (Rados13) [inline assets/js/components/ControlPanel.tsx:204]: I would move it also to file which will return fontColor.

File: assets/js/components/ControlPanel.tsx [hunk 11]
```diff
@@ -147,13 +192,36 @@ const ControlPanel = ({ clientName, webrtc, playerCallback }: ControlPanelProps)
...
+              />
+            }
+            onClick={() => {
+              eventChannel.push("presenter_remove", { presenter: clientName });
+              setMode("hls");
+            }}
```
Comment (Rados13) [inline assets/js/components/ControlPanel.tsx:210]: I would move it to seperate function, outside of html elements.

**Review by Rados13 — CHANGES_REQUESTED** [review]

File: assets/js/utils/modePanelUtils.ts (hunk 1 above)
Comment (LVala) [reply assets/js/utils/modePanelUtils.ts:20]: Fixed.

File: assets/js/components/StreamArea.tsx (hunk 3 above)
Comment (LVala) [reply assets/js/components/StreamArea.tsx:22]: Fixed (tho I'm not the one that wrote that :( ).

File: lib/membrane_live/event.ex (hunk 5 above)
Comment (LVala) [reply lib/membrane_live/event.ex:172]: Explained.

File: assets/js/components/ParticipantsList.tsx (hunk 6 above)
Comment (LVala) [reply assets/js/components/ParticipantsList.tsx:40]: Fixed.

File: assets/js/components/ParticipantsList.tsx (hunk 7 above)
Comment (LVala) [reply assets/js/components/ParticipantsList.tsx:37]: Fixed.

File: assets/js/components/ControlPanel.tsx (hunk 10 above)
Comment (LVala) [reply assets/js/components/ControlPanel.tsx:204]: Fixed.

File: assets/js/components/ControlPanel.tsx (hunk 11 above)
Comment (LVala) [reply assets/js/components/ControlPanel.tsx:210]: Fixed.

File: assets/js/components/StreamArea.tsx (hunk 4 above)
Comment (LVala) [reply assets/js/components/StreamArea.tsx]: Explained.

File: assets/js/components/ControlPanel.tsx [hunk 12]
```diff
@@ -36,6 +36,7 @@ import {
   SourceType,
 } from "../utils/rtcUtils";
 import { Mode } from "./StreamArea";
+import { getFontColor } from "./ParticipantsList";
```
Comment (Rados13) [inline assets/js/components/ControlPanel.tsx:39]: It looks a little bit strange that you import `getFontColor` from `ParticipantsList`, maybe we could create some utils file for this?

**Review by Rados13 — APPROVED** [review]

File: assets/js/components/ControlPanel.tsx (hunk 12 above)
Comment (LVala) [reply assets/js/components/ControlPanel.tsx:39]: Fixed.

**Review by Rados13 — APPROVED** [review]

**Pull request merged**